src/renderer/src/routes/modals/PrepareHardwareWalletMigrationNeo3/index.tsx

```diff
@@ -0,0 +1,254 @@
+import { useModalNavigate, useModalState } from '@renderer/hooks/useModalRouter'
+import { CenterModalLayout } from '@renderer/layouts/CenterModal'
+import { THardwareWalletInfo } from '@shared/@types/ipc'
+import { IAccountState } from '@shared/@types/store'
+
+import { PrepareHardwareWalletAccounts } from './PrepareHardwareWalletAccounts'
+import { PrepareHardwareWalletAddressItem } from './PrepareHardwareWalletAddressItem'
+import { PrepareHardwareWalletConnection } from './PrepareHardwareWalletConnection'
+import { PrepareHardwareWalletContainer } from './PrepareHardwareWalletContainer'
+import { PrepareHardwareWalletTipInfo } from './PrepareHardwareWalletTipInfo'
+
+type TLocationState = {
+  account: IAccountState
+}
+
+type TActionsData = {
+  isGoingToMigrationNeo3: boolean
+  isNeo3StepDone: boolean
+  isAddingNeo3Account: boolean
+  neo3Account: Account<'neo3'> | null
+  neo3WalletInfo: THardwareWalletInfo | null
+}
+
+export const PrepareHardwareWalletMigrationNeo3Modal = () => {
+  const { t } = useTranslation('modals', { keyPrefix: 'prepareHardwareWalletMigrationNeo3' })
+  const { account } = useModalState<TLocationState>()
+  const { modalErase } = useModalNavigate()
+  const balanceQuery = useBalance(account)
+  const { canMigrateToNeo3 } = useMigrationNeo3Validations({ account })
+  const { createHardwareWallet } = useHardwareWalletActions()
+  const navigate = useNavigate()
+
+  const {
+    actionData: { isGoingToMigrationNeo3, isNeo3StepDone, isAddingNeo3Account, neo3Account, neo3WalletInfo },
+    actionState,
+    setData,
+  } = useActions<TActionsData>({
+    isGoingToMigrationNeo3: false,
+    isNeo3StepDone: false,
+    isAddingNeo3Account: false,
+    neo3Account: null,
+    neo3WalletInfo: null,
+  })
+
+  const neo3ConnectHardwareWallet = useConnectHardwareWallet(
+    async walletsInfo => {
+      setData({ neo3WalletInfo: walletsInfo.find(walletInfo => walletInfo.blockchain === 'neo3') })
+    },
+    {
+      canConnect: !isNeo3StepDone,
+      afterConnectionMs: 0,
+      validateBeforeConnection: walletsInfo => {
+        const neo3WalletInfo = walletsInfo.find(({ blockchain }) => blockchain === 'neo3')
+
+        return !!neo3WalletInfo && neo3WalletInfo.accounts.length > 0
+      },
+    }
+  )
+
+  const connectHardwareWallet = useConnectHardwareWallet(
+    async walletsInfo => {
+      await createHardwareWallet(walletsInfo)
+    },
+    {
+      canConnect: isNeo3StepDone,
+      afterConnectionMs: 0,
+      validateBeforeConnection: walletsInfo => {
+        const walletInfo = walletsInfo.find(({ blockchain }) => blockchain === 'neoLegacy')
+
+        return !!walletInfo && walletInfo.accounts.some(AccountHelper.predicate(account))
+      },
+    }
+  )
+
+  const currentStep = useMemo(() => {
+    if (isGoingToMigrationNeo3) return 6
+    if (isNeo3StepDone) return 4
+    if (neo3WalletInfo) return 3
+
+    return 2
+  }, [isNeo3StepDone, isGoingToMigrationNeo3, neo3WalletInfo])
```

Comment:
It does not need to be a useMemo
```suggestion
  const currentStep = match({isGoingToMigrationNeo3, isNeo3StepDone, neo3WalletInfo})
  .with({isGoingToMigrationNeo3: true}, () => 6)
  .with({isNeo3StepDone: true}, () => 4)
  .with({neo3WalletInfo}, () => 3)
  .otherwise(() => 2)
```